### PR TITLE
Carbon black enterprise live response memdump

### PR DESCRIPTION
**Pull request by uppysec**
## Status
Ready

## Description
Fixed Memdump command in existing integration

## Screenshots
Paste here any images that will help the reviewer

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------

## Required version of Demisto
x.x.x

## Does it break backward compatibility?
No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Dependencies
None
## Additional changes
Fixed Memdump command

### CODE REVIEW

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 1]
```diff
@@ -628,7 +628,7 @@ script:
     }
 
     function createCommandAndWait() {
-        let result = createCommandRequest(args.session, args.name, args.timeout, args.object, args.compress, args.wait,
+        let result = createCommandRequest(args.session, args.name, args.timeout, args.object, args.commpress, args.wait,
```
Comment (avidan-H) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: Why the changes from `commpress` to `compress`? (I know that `commpress` is not the correct spelling of the word but it's what is used in Carbon Black's API)

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 2]
```diff
@@ -1325,13 +1325,13 @@ script:
...
+    function memdumpRequest(sessionId, path) {
         let queryParams = {
             name: 'memdump',
-            object: path
+            object: path,
         };
-        if (commpress) {
-            queryParams.commpress = commpress;
+        if (args.compress) {
+            queryParams.compress = true;
```
Comment (avidan-H) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: I believe this change is incorrect as the CB API has it as `commpress`. Looking at the API documentation [here](https://developer.carbonblack.com/reference/enterprise-response/5.1/live-response-api/#memdump).

Comment (uppysec) [reply Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: Hi, the API documentation is incorrect, it looks like a typo.
Here is the result of running the updated command:
image

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml (hunk 1 above)
Comment (uppysec) [reply Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: Fixed the typo in latest commit. Thanks

**Comment by ronykoz** [comment]
@uppysec sorry for the delayed response, @avidan-H will continue is review shortly

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 3]
```diff
@@ -1325,13 +1325,13 @@ script:
         return commandEntries['create process'](command);
     }
 
-    function memdumpRequest(sessionId, path, commpress) {
+    function memdumpRequest(sessionId, path) {
```
Comment (avidan-H) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]:
```suggestion
    function memdumpRequest(sessionId, path, compress) {
```

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 4]
```diff
@@ -1325,13 +1325,13 @@ script:
...
         let queryParams = {
             name: 'memdump',
-            object: path
+            object: path,
```
Comment (avidan-H) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]:
```suggestion
            object: path
```

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 5]
```diff
@@ -1325,13 +1325,13 @@ script:
...
         };
-        if (commpress) {
-            queryParams.commpress = commpress;
+        if (args.compress) {
```
Comment (avidan-H) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]:
```suggestion
        if (compress === 'true') {
```

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 6]
```diff
@@ -1341,17 +1341,14 @@ script:
         }
     }
 
-    function memedump() {
-        if (commpress) {
-            args.commpress = args.commpress === 'yes';
-        }
-        let result = memdumpRequest(args['session'], args.path, args.commpress);
+    function memdump() {
+        let result = memdumpRequest(args['session'], args.path);
```
Comment (avidan-H) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]:
```suggestion
        let result = memdumpRequest(args['session'], args.path, args.compress);
```

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 7]
```diff
@@ -1632,7 +1629,7 @@ script:
                 return directoryList();
             case 'cb-process-execute':
                 return processExec();
-            case 'cb-memdeump':
+            case 'cb-memdump':
```
Comment (avidan-H) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]:
```suggestion
            case 'cb-memdeump':
```

Comment (avidan-H) [reply Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: revert this back to the typo spelling in order to maintain backward compatibility.

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 8]
```diff
@@ -2088,7 +2088,7 @@ script:
     - contextPath: CbLiveResponse.Commands.ProcessID
       description: The pid of the executed process.
     description: Run executable on the endpoint.
-  - name: cb-memdeump
+  - name: cb-memdump
```
Comment (avidan-H) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]:
```suggestion
  - name: cb-memdeump
```

Comment (avidan-H) [reply Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: revert back to typo spelling for backward compatibility reasons.

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 9]
```diff
@@ -2665,5 +2667,3 @@ script:
       description: The destination path of the file.
     description: Save a file to a specific path on the endpoint.
   runonce: false
```
Comment (avidan-H) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: please add back the tests field and value that you deleted from the yaml

File: Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml [hunk 10]
```diff
@@ -2110,6 +2107,11 @@ script:
       - "yes"
       - "no"
       description: Cancel the command if still 'pending' after timeout.
+    - name: compress
```
Comment (avidan-H) [inline Integrations/integration-Carbon_Black_Enterprise_Live_Response.yml]: need to revert this change to preserve backward compatibility.

**Comment by avidan-H** [comment]
@uppysec also please sign the CLA when you get a chance

**Comment by avidan-H** [comment]
Closing due to inactivity. Opened an issue with the details as well as a link to this PR.
